### ISSUE TITLE
slide generators

### ISSUE THREAD

**Issue by blahah** [slidewinder/app#6]
in addition to being able to create cards from scratch, and import them from external slide sources, a third option should be available: slide generators.

These work by using a template and some data source, which could be a remote API or a user-provided file, to generate slides.

There should be a general framework for creating and using generators.

**Comment by TransGirlCodes** [comment]
Perhaps a sub goal for this is getting the template slides concept completed.

**Comment by TransGirlCodes** [comment]
Related but separate organisation level question: Should the generators be a separate sidewinder project/repo? We've got the library/command line repo, the app repo. Is generators, and something - some server or bot or daemon that runs them and adds the slides to a database - a separate application to the lib/cmdline and the app? Perhaps we are approaching a holy sharing trinity.

**Comment by blahah** [comment]
Good point about the template setup, I think that should be a separate issue.

**Comment by blahah** [comment]
potentially each generator could be a separate project to allow people to pick and choose which ones they want to install, like installing atom packages or something

**Comment by TransGirlCodes** [comment]
Oh God yes! It would be so cool to do `slidewinder generators list` or
`sidewinder generators run` or this kind of thing. Where it can install,
remove, start, and stop them!
